Algorithms/Encoders/SoundexEncoder.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Linq;
+using System.Text;
+using System.Globalization;
+
+namespace Algorithms.Encoders
+{
+    /// <summary>
+    /// Class for Soundex encoding strings.
+    /// </summary>
+    public class SoundexEncoder
+    {
+        /// <summary>
+        /// Encodes a string using the Soundex Algorithm
+        /// </summary>
+        /// <param name="text">The string to encode</param>
+        /// <returns>The Soundex encoded string (one uppercase character and three digits)</returns>
+        public string Encode(string text)
+        {
+            StringBuilder soundex = new StringBuilder($"{text.ToUpper(CultureInfo.CurrentCulture)[0]}");
+            text = string.Join("", text.Select(ch => {
+                switch(char.ToLower(ch))
+                {
+                    //remove aeiouyhw (map to zero)
+                    case 'a':
+                    case 'e':
+                    case 'i':
+                    case 'o':
+                    case 'u':
+                    case 'y':
+                        return '0';
+
+                    case 'h':
+                    case 'w':
+                        return '8';
```

Comment:
Can we just remove h and w? This will simplify checks downstream. If you remove them, then adjacent duplicates can be merged